spec/integration/test.js

```diff
@@ -192,6 +194,47 @@ describer('notification api with a live service', function () {
       });
     });
 
+    it('get a letter pdf', (done) => {
+      should.exist(letterNotificationId)
+      pdf_contents = fs.readFileSync('./spec/integration/test_files/one_page_pdf.pdf');
+      var count = 0
+      // make sure test is closed
+      const endTest = (err) => {
+        if (err !== undefined) {
+          done(err);
+        } else {
+          done();
+        }
+      };
+      // it takes a while for the pdf for a freshly sent letter to become ready, so we need to retry the promise
+      // a few times, and apply delay in between the attempts. Since our function returns a promise,
+      // and it's asynchronous, we cannot use a regular loop to do that. That's why we envelop our promise in a
+      // function which we call up to 7 times or until the promise is resolved.
+      const tryClient = () => {
+        return notifyClient.getPdfForLetterNotification(letterNotificationId)
+        .then((file_buffer) => {
+          var fileBuffer = file_buffer
+          expect(pdf_contents).to.equalBytes(file_buffer);
+          endTest();
+        })
+        .catch((err) => {
+          if (err.constructor.name === 'AssertionError') {
+            endTest(err)
```

Comment:
here we know `err` isn't undefined since we've checked its constructor, so we can just call `done(err);`
```suggestion
            done(err);
```

Reply:
also i'm surprised that we're getting `AssertionError`s inside this catch but i guess i just don't understand promises enough 🤷‍♂